vmm/src/vstate.rs

```diff
@@ -92,13 +94,24 @@ impl Vm {
     pub fn new(kvm: &Kvm) -> Result<Self> {
         //create fd for interacting with kvm-vm specific functions
         let vm_fd = kvm.create_vm().map_err(Error::VmFd)?;
-
+        #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+        let cpuid = kvm
+            .get_supported_cpuid(MAX_KVM_CPUID_ENTRIES)
+            .map_err(Error::VmFd)?;
         Ok(Vm {
             fd: vm_fd,
+            #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+            cpuid,
             guest_mem: None,
         })
     }
 
+    /// Returns a clone of the supported `CpuId` for this Vm.
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub fn get_supported_cpuid(&self) -> CpuId {
```

Comment:
Is there any reason why the variable is called `cpuid` and the getter is called `get_supported_cpuid` ?

Reply:
It is called get_supported_cpuid because it actually returns a clone of the supported cpuid for this VM. I can rename the field in Vm to also be called supported_cpuid. Would that make sense?

Reply:
Done.